[jellywaiyan] iP

jellywaiyan opened the pull request:
# Jelly
> Hello :o

### Jelly frees your mind of having to remember things you need to do. It's,

1. Text-based
2. Easy to learn
3. ~~FAST~~ MAYBE FAST to use :o

### **All you need to do is,**

- Download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
- Double-click it.
- Add your tasks.
- Let it manage your tasks for you 😮 

And it is ***FREE***!

### Features:
- [x] Managing tasks
- [ ] Managing deadlines
- [ ] Reminders

If you are a java programmer, you can use it to practice Java too. Here's the `main` method:
```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

jellywaiyan committed:
Level-0

jellywaiyan committed:
Leve-1

jellywaiyan committed:
Level-2

jellywaiyan committed:
Level-3

jellywaiyan committed:
Level-4

jellywaiyan committed:
A-TextUiTesting

jellywaiyan committed:
Level-5

jellywaiyan committed:
Level-6

jellywaiyan committed:
Level-6

jellywaiyan committed:
Level-7

jellywaiyan committed:
Level-7v2

jellywaiyan committed:
Merge branch 'branch-Level-7'

jellywaiyan committed:
Level-8 deadline done, event left to do

jellywaiyan committed:
Level-8

jellywaiyan committed:
Merge branch 'branch-Level-8'

jellywaiyan committed:
Add Storage, TaskList, Ui

jellywaiyan committed:
A-MoreOOP

jellywaiyan committed:
Merge remote-tracking branch 'jellywaiyan/A-MoreOOP'

jellywaiyan committed:
A-Packages

jellywaiyan committed:
Merge remote-tracking branch 'jellywaiyan/A-Packages'

jellywaiyan committed:
Merge remote-tracking branch 'jellywaiyan/master' into add-gradle-support

jellywaiyan committed:
A-Gradle

jellywaiyan committed:
A-JUnit

jellywaiyan committed:
Merge remote-tracking branch 'jellywaiyan/A-JUnit'

jellywaiyan committed:
A-Jar

jellywaiyan committed:
Merge tag 'A-Jar'

jellywaiyan committed:
Added javadocs to most methods/classes

jellywaiyan committed:
A-CodingStandard

jellywaiyan committed:
Level-9

jellywaiyan committed:
Merge remote-tracking branch 'jellywaiyan/A-JavaDoc'

jellywaiyan committed:
Merge remote-tracking branch 'jellywaiyan/A-CodingStandard'

# Conflicts:
#	src/main/java/Jelly/main/Storage.java
#	src/main/java/Jelly/main/TaskList.java
#	src/main/java/Jelly/main/Ui.java
#	src/main/java/Jelly/task/Deadline.java
#	src/main/java/Jelly/task/Event.java
#	src/main/java/Jelly/task/Todo.java

jellywaiyan committed:
Merge remote-tracking branch 'jellywaiyan/Level-9'

# Conflicts:
#	src/main/java/Jelly/main/Jelly.java
#	src/main/java/Jelly/main/Ui.java

yiwen101 commented on src/main/java/Jelly/task/Event.java:
```diff
@@ -54,6 +58,7 @@ public String toString() {
     }
+
     @Override
     public String writeToFile() {
```
Could be better named. 
I do not expect"writeToFile" to be a method that returns the String to be written

yiwen101 commented on src/main/java/Jelly/main/Jelly.java:
```diff
@@ -24,11 +32,21 @@ private Jelly(String filePath) throws JellyException {
+    /**
+     * The main initialiser for the Jelly Chat bot.
+     *
+     * @param args
```
I do not think  "@param args" help much
And I believe that you do not need to document every method as well.

yiwen101 commented on src/main/java/Jelly/main/Ui.java:
```diff
@@ -5,42 +5,81 @@
+    /**
+     * Prints out a welcome message when the Chat Bot is booted up.
+     */
     public void startUpMessage() {
```
sounds like a noun

yiwen101 commented on src/main/java/Jelly/main/Ui.java:
```diff
@@ -5,42 +5,81 @@
+     * @param addedTask The task that was added.
+     * @param noOfTasks The total number of tasks in the list after adding.
+     */
+    public void addedTaskMessage(Task addedTask, int noOfTasks) {
```
noun

yiwen101 commented on src/main/java/Jelly/main/Ui.java:
```diff
@@ -5,42 +5,81 @@
+    /**
+     * Displays a final message to the user before the Chat Bot shuts down.
+     */
     public void byeMessage() {
```
noun

yiwen101 commented on src/main/java/Jelly/task/Task.java:
```diff
@@ -1,39 +1,70 @@
+     * Should not happen, as tasks should be either one of these: todo, deadline or event.
+     * @return Error message.
+     */
     public String writeToFile() {
```
if this should not happen, why not throw exception when called?

yiwen101 reviewed (commented):
Looks good
Quite comprehensive documentation
Well designed

ElginTZM commented on src/main/java/Jelly/commands/EventCommand.java:
```diff
@@ -0,0 +1,30 @@
+
+    private String fromWhen;
+
+    private String toWhen;
```
Perhaps can be grouped together

ElginTZM commented on src/main/java/Jelly/commands/DeleteCommand.java:
```diff
@@ -0,0 +1,26 @@
+        this.index = index;
+    }
+
+    @Override
+    public void execute(TaskList taskList, Ui ui, Storage storage) {
+        if (index <= 0 || index > 100 || index > taskList.size()) {
+            System.out.println("Invalid input");
+        } else if (taskList.get(index - 1) != null) {
+            ui.deleteMessage(taskList.get(index - 1), taskList.size() - 1);
+            taskList.delete(index - 1);
+        } else {
+            System.out.println("Invalid input");
+        }
+    }
```
Not really a coding standard issue but perhaps you can try guard clauses to improve readability?

```java
@Override
    public void execute(TaskList taskList, Ui ui, Storage storage) {
        if (index <= 0 || index > 100 || index > taskList.size()) {
            System.out.println("Invalid input");
            return;
        }
        
        if (taskList.get(index - 1) != null) {
            ui.deleteMessage(taskList.get(index - 1), taskList.size() - 1);
            taskList.delete(index - 1);
            return;
        }
        
        System.out.println("Invalid input");
    }
```

ElginTZM commented on src/main/java/Jelly/commands/MarkCommand.java:
```diff
@@ -0,0 +1,29 @@
+        this.index = index;
+    }
+
+    @Override
+    public void execute(TaskList taskList, Ui ui, Storage storage) {
+        if (index <= 0 || index > 100) {
+            System.out.println("Invalid input");
+        } else if (taskList.get(index - 1) != null) {
+            if (taskList.get(index - 1).getTaskStatus() == "X") {
+                System.out.println("Uh, it appears that you've finished this task o.o");
+            } else {
+                taskList.get(index - 1).markAsDone();
+                System.out.println("Good job! I've marked this task as done :)");
+            }
+        } else {
+            System.out.println("Invalid input");
+        }
+    }
```
Similar to above comment, not really a coding standard thing, but perhaps you can use guard clauses to prevent nesting of if else statements to improve readability?

```java
@Override
    public void execute(TaskList taskList, Ui ui, Storage storage) {
        if (index <= 0 || index > 100) {
            System.out.println("Invalid input");
            return;
        }

        // Check for the case where it is null to remove 1 level of nesting
        if (taskList.get(index - 1) == null) {
            System.out.println("Invalid input");
            return;
        }
        
        if (taskList.get(index - 1).getTaskStatus() == "X") {
            System.out.println("Uh, it appears that you've finished this task o.o");
            return;
        }

        taskList.get(index - 1).markAsDone();
        System.out.println("Good job! I've marked this task as done :)");
    }
```

ElginTZM commented on src/test/java/Jelly/main/TaskListTest.java:
```diff
@@ -0,0 +1,29 @@
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertFalse;
```
Perhaps you can separate out the imports?

ElginTZM commented on src/test/java/Jelly/main/StorageTest.java:
```diff
@@ -0,0 +1,44 @@
+
+public class StorageTest {
+    @Test
+    public void startUpTest() throws IOException, JellyException {
```
Test methods name can utilize another convention for naming them: `featureUnderTest_testScenario_expectedBehavior()`

ElginTZM reviewed (commented):
Looks great! No major coding standard issues, just some minor issues with how you name some variables like the other reviewer have mentioned

jellywaiyan committed:
A-CheckStyle

jellywaiyan committed:
A-CheckStyle

jellywaiyan committed:
Merge remote-tracking branch 'jellywaiyan/A-CheckStyle'

jellywaiyan committed:
Level-10

jellywaiyan committed:
Merge remote-tracking branch 'jellywaiyan/branch-Level-10'

jellywaiyan committed:
Improve code quality by making methods more efficient and readable.

Delete command: Save to file after deleting
Mark command: Save to file after marking a task as done
Unmark command: Save to file after marking a task as not done

Previously, these commands did not save to file, which caused the file to be outdated.

Let's,
*add a saveToFile method after completing the command

jellywaiyan committed:
Merge pull request #2 from jellywaiyan/branch-A-CodeQuality

Improve code quality by making methods more efficient and readable.

jellywaiyan committed:
Current implementation does not use assertions.

Assert feature allows the documentation of important assumptions that should hold at various points in the code.

Let's,
*add assertions in Task, Parser,DeleteCommand, MarkCommand, UnmarkCommand.

jellywaiyan committed:
Small change

jellywaiyan committed:
Merge branch 'master' into A-Assertions

jellywaiyan committed:
Merge A-Assertions

jellywaiyan committed:
Merge pull request #3 from jellywaiyan/A-Assertions

Current implementation does not use assertions.

jellywaiyan committed:
Fix checkstyle

jellywaiyan committed:
Add priority to tasks

jellywaiyan committed:
Add priority

jellywaiyan committed:
Merge remote-tracking branch 'jellywaiyan/C-Priority'

jellywaiyan committed:
Change README, add Ui.png

jellywaiyan committed:
Configure jar file
